lib/error.js

```diff
@@ -0,0 +1,36 @@
+/** takes an error object and serializes it to a plan object */
+function serializedError (input) {
+  if (!(input instanceof Error)) {
+    const error = new Error('attempted to serialize a non-error object')
+    return serializedError(error)
+  }
```

Comment:
Occasionally someone (or their AI assistant) will mess up and throw a bare string, especially if we're not using TypeScript.

Let's check for that and preserve the string. Maybe use `${input} (string, not error object)` for `message`

Reply:
added typeof / constructor name

Reply:
Adding `typeof` doesn't address my comment.

Say I'm updating the code below. When I switch to using safeError, I think it would be good if "Disk is full" was still logged. Seeing the intended error message could be useful in the case of an incident.

```diff
try {
   // Oops, I left off new Error
   throw 'Disk is full'
} catch (err) {
-  console.error(err)
+  console.error(safeError(err))
}
```

Reply:
sorry I forgot to address logging the direct value. I think logging the value of an unknown type is harmful, what if the string is very large? the stack trace alone should be good enough to track it back to where it happened, and to remediate it.

Reply:
We may not get a useful stack, but I can understand erring on the side of safety.

Optional: We could truncate the value if it's a string.

Reply:
done